File: advent-of-code-2020/solution01.py
entries = []

with open('input01.txt', encoding='utf8') as f:
    for line in f:
        line = line.strip()
        entries.append(int(line))

# part 1
found = False
for i in entries:
    if found:
        break
    for j in entries:
        if i == j:
            continue
        if i + j == 2020:
            print(i, j, i*j)
            found = True
# part 2
found = False
for i in entries:
    if found:
        break

    for j in entries:
        if i == j:
            continue
        elif found:
            break

        for k in entries:
            if i == k:
                continue

            if i + j + k == 2020:
                print(i, j, k, i*j*k)
                found = True

File: advent-of-code-2020/solution01.cc
#include <vector>
#include <string>
#include <iostream>
#include <fstream>

using std::vector;


int main() {
    vector<int> entries;
    std::ifstream myfile;
    std::string line;

    myfile.open("input01.txt");

    while (getline(myfile,line))
        {
          entries.push_back(std::stoi(line));
        }
    myfile.close();

    bool found = false;
    for(size_t i = 0; i != entries.size(); i++) {
        if (found) { 
            break;
        }
        for(size_t j = 0; j != entries.size(); j++) {
            if (entries[i] == entries[j]) {
                continue;
            }
            else if (found) {
                break;
            }
            if (entries[i] + entries[j] == 2020) {
                std::cout << entries[i] << " " << entries[j] << " " << entries[i]*entries[j] << std::endl;
                found = true;
            }
        }
    }
}